File: copilot/aspnet/securable/Models/ShareLink.cs
namespace LooseNotes.Models;

public class ShareLink
{
    public int Id { get; set; }
    public int NoteId { get; set; }
    public string Token { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool IsActive { get; set; } = true;

    public Note? Note { get; set; }
}

File: claude/node/securable/src/models/ShareLink.js
'use strict';

const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const ShareLink = sequelize.define(
    'ShareLink',
    {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
      },
      noteId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: { model: 'notes', key: 'id' },
      },
      token: {
        type: DataTypes.STRING(128),
        allowNull: false,
        unique: true,
      },
      isActive: {
        type: DataTypes.BOOLEAN,
        allowNull: false,
        defaultValue: true,
      },
    },
    {
      tableName: 'share_links',
      indexes: [{ fields: ['token'] }, { fields: ['note_id'] }],
    }
  );

  ShareLink.associate = (models) => {
    ShareLink.belongsTo(models.Note, { foreignKey: 'noteId', as: 'note' });
  };

  return ShareLink;
};
